tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -279,3 +280,166 @@ def test_45498(self):
+                    "user.")
+
+        LOGGER.info("Step 5: Create IAM user and perform IOs after soft-failure.")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-45501-1'
+        self.s3_clean.update(users)
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
+                                                    nclients=2, nsamples=2, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 5: Created IAM user and performed IOs after soft-failure.")
+        LOGGER.info("ENDED: Verify IAM user creation & IOs before and after soft-failures.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45502")
+    def test_45502(self):
+        """
+        Verify IAM user creation/deletion and get IAM user list while soft-failure in loop.
+        """
+        LOGGER.info("STARTED: Verify IAM user creation/deletion and get IAM user list while "
+                    "soft-failure in loop.")
+        LOGGER.info("Precondition: Create %s iam users for deletion", self.num_users)
+        del_users = self.mgnt_ops.create_account_users(nusers=self.num_users)
+
+        LOGGER.info("Step 1: Create multiple IAM users.")
+        new_users = self.mgnt_ops.create_account_users(nusers=10)
+        new_users_list = list(new_users.keys())
+        LOGGER.info("Step 1: Created multiple IAM users.")
+
+        LOGGER.info("Step 2: Start IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verify existing IAM users persist.")
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        event = threading.Event()
+        args = {'user_crud': True, 'num_users': self.num_users,
+                'output': iam_output, 'del_users_dict': del_users, 'header': self.header}
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event,), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        LOGGER.info("Starting background threads for IAM user creation/deletion.")
+        thr_iam.start()
+        for failure in range(HA_CFG["common_params"]["short_loop"]):
+            pod = secrets.SystemRandom().choice(pod_list)
```

Comment:
can we check if not same pod is repeated and loop around number of pods instead of loop number?
So in case we have only 2 control pods, we can loop around only 2 times for 2 diff pods??

Reply:
> can we check if not same pod is repeated and loop around number of pods instead of loop number? So in case we have only 2 control pods, we can loop around only 2 times for 2 diff pods??

We want to check random pod deletion multiple times. We already have TEST-45498 with this scenario.